internal/venice-common/src/main/java/com/linkedin/venice/meta/Version.java

```diff
@@ -379,14 +402,18 @@ static String parseStoreFromRealTimeTopic(String kafkaTopic) {
     if (!isRealTimeTopic(kafkaTopic)) {
       throw new VeniceException("Kafka topic: " + kafkaTopic + " is not a real-time topic");
     }
-
-    int lastIndexOfVersionSeparator = kafkaTopic.lastIndexOf(VERSION_SEPARATOR);
-    // we only care about the prefix, so providing topic in place of store should work
-    if (lastIndexOfVersionSeparator != -1 && Utils.isRTVersioningApplicable(kafkaTopic)) {
-      return kafkaTopic.substring(0, lastIndexOfVersionSeparator);
+    String topicWithoutRTVersionSuffix = kafkaTopic;
+    try {
+      topicWithoutRTVersionSuffix = removeRTVersionSuffix(kafkaTopic);
+    } catch (IllegalArgumentException e) {
+      // nothing to do, it could be an old styled real time topic
```

Comment:
With the updated code, the comments is not true anymore, right? Is it right to swallow it here? Do you think we can concert it into a Venice Exception and throw it out or